### PR TITLE
Feat/claim logs

### PR DESCRIPTION
**Pull request by patryk-kochanski**

### CODE REVIEW

File: packages/origin-247-claim/migrations/1628061005210-ClaimLogs.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export class ClaimLogs1628061005210 implements MigrationInterface {
+    name = 'ClaimLogs1628061005210';
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            `CREATE TABLE "excess_generation" ("id" SERIAL NOT NULL, "createdAt" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "updatedAt" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "generatorId" text NOT NULL, "volume" text NOT NULL, "from" TIMESTAMP WITH TIME ZONE NOT NULL, "to" TIMESTAMP WITH TIME ZONE NOT NULL, "generatorMetadata" text NOT NULL, CONSTRAINT "PK_148501091c857546abf2b5cd7c5" PRIMARY KEY ("id"))`
+        );
+        await queryRunner.query(
+            `CREATE TABLE "leftover_consumption" ("id" SERIAL NOT NULL, "createdAt" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "updatedAt" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "consumerId" text NOT NULL, "volume" text NOT NULL, "from" TIMESTAMP WITH TIME ZONE NOT NULL, "to" TIMESTAMP WITH TIME ZONE NOT NULL, "consumerMetadata" text NOT NULL, CONSTRAINT "PK_d85f411d0ae242db81426c8eb29" PRIMARY KEY ("id"))`
```
Comment (soanvig) [inline packages/origin-247-claim/migrations/1628061005210-ClaimLogs.ts:11]: TBH I would use name `match_leftover_consumption`

File: packages/origin-247-claim/src/repositories/ExcessGeneration/ExcessGeneration.entity.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    updatedAt: Date;
+
+    @Column({ type: 'text' })
+    generatorId: string;
+
+    @Column({ type: 'text' })
+    volume: string; //BigNumber
+
+    @Column({ type: 'timestamptz' })
+    from: Date;
```
Comment (soanvig) [inline packages/origin-247-claim/src/repositories/ExcessGeneration/ExcessGeneration.entity.ts:29]: Now I wonder - do we really need this `from`?

File: packages/origin-247-claim/src/repositories/ExcessGeneration/ExcessGeneration.entity.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Column({ type: 'timestamptz' })
+    to: Date;
```
Comment (soanvig) [inline packages/origin-247-claim/src/repositories/ExcessGeneration/ExcessGeneration.entity.ts:32]: And if we remove `from` we could simply put `timestamp` or anything here

File: packages/origin-247-claim/src/repositories/MatchResult/MatchResultPostgres.repository.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                generatorId: In(findOptions.generatorIds),
+                from: Between(findOptions.from, findOptions.to),
+                to: Between(findOptions.from, findOptions.to)
+            }
+        });
+    }
+
+    public async findGrouped(
+        findOptions: FindOptions,
+        groupOptions: Entity[]
```
Comment (soanvig) [inline packages/origin-247-claim/src/repositories/MatchResult/MatchResultPostgres.repository.ts:45]: TBH I don't understand how this API works here.

**Review by soanvig — CHANGES_REQUESTED** [review]

File: packages/origin-247-claim/src/repositories/MatchResult/MatchResultPostgres.repository.ts (hunk 4 above)
Comment (patryk-kochanski) [reply packages/origin-247-claim/src/repositories/MatchResult/MatchResultPostgres.repository.ts:45]: Maybe changing the names to filterBy and groupBy would make more sense?
findOptions is the data for **WHERE**
` 
example:
    {
        consumerIds: ['consumerOne', 'consumerTwo'],
        generatorIds: ['generatorOne', 'generatorTwo', 'generatorThree'],
        from: new Date('2021-08-01T10:00:00.000Z'),
        to: new Date('2021-08-01T12:00:00.000Z')
    }
`
groupOptions is the data for **GROUP BY**
`
examples:
groupOptions = [Entity.Generator];
groupOptions = [[Entity.Generator, Entity.Consumer];
groupOptions = [[Entity.Consumer, Entity.Generator];
`

Comment (soanvig) [reply packages/origin-247-claim/src/repositories/MatchResult/MatchResultPostgres.repository.ts:45]: Oh, so it's wrong data typa I think. It should be `[Entity] | [Entity, Entity]` ?

**Review by soanvig — APPROVED** [review]

**Pull request merged**